cmd/image-builder/manifest.go

```diff
@@ -39,11 +39,13 @@ func sbomWriter(outputDir, filename string, content io.Reader) error {
 	if err != nil {
 		return err
 	}
+	// ensure we do not leak FDs if the function returns prematurely
 	defer f.Close()
```

Comment:
But this will trigger the linter again, right?
why not go for the slightly more bulky but complete
```
defer func() { err = errors.Join(err, f.Close()) }()
```
otherwise the `// nolint:errcheck` would be missing

Reply:
In ibcli we have no linter complaining yet so I did not add this. Once we add the updated linter this (and more places) will need to change.

We could use the  `errors.Join`. I slightly prefer the other one because:
a) the rules about named return vars in go can be a bit confusing (i.e. what shadows what)
b) the `return f.Close()` at the end is slightly more linear, i.e. its imeediately obvious that the close error is returned here
but its only a slight preference, the extra comment in this version is also kinda annoying.

Reply:
… this way we are doing a double-close - although the error is ignored - not sure if this is the most elegant solution - but it will probably catch many important cases now

Reply:
Yes, in the general (successful) case, there will always be a deferred close which produces an error that is ignored.